src/germanNumber.ts

```diff
@@ -21,9 +21,14 @@ export function formatGermanNumber(
     return defaultValue ?? '';
   }
 
+  let { maximumFractionDigits } = localeOptions;
+  if (!maximumFractionDigits || maximumFractionDigits > 6) {
+    maximumFractionDigits = 6;
```

Comment:
I would expect `formatGermanNumber(123.123456789, { maximumFractionDigits: 7 }).toBe('123,1234567')`, however this will limit it to `6`. I think that not allowing values above 6 at all and silently changing them is undesirable and unexpected.